### PR TITLE
Closes Issue: 701: Not exposed desktop mode in browser session

### LINKED ISSUE #701: Desktop mode not exposed in browser-session
c854509a704af24738c3097611547793a25193ee introduced desktop mode. However there's no flag on `Session` for the application to observe.

* In `Session` we want to add a property for desktop mode.
* In `EngineObserver` we want to implement `onDesktopModeEnabled` and update the field in `Session`

### PR DESCRIPTION
**Pull request by Amejia481**
Closes #701: Exposing browser-session Desktop mode.

* Adding desktopMode property to Session

* Adding onDesktopModeEnabled in Session

* Renaming RequestDesktopSite to RequestDesktopSiteUseCase to follow
  the convention

* Implementing onDesktopModeEnabled in EngineObserver

* I kept the function name setDesktopMode instead of changing it to
  desktopMode, to follow the existed convention (all function are verbs)
  and because it denotes the action of changing the value

### CODE REVIEW

File: docs/api/0.21/feature-session/index-outline.html [hunk 1]
```diff
@@ -157,7 +157,7 @@
...
...spari/Projects/Mozilla/android-components/build/javadoc/feature-session/index"><a href="mozilla.components.feature.session/-session-use-cases/-request-desktop-site/invoke.html"><span class="keyword">fun </span><span class="identifier">invoke</span><span class="symbol">(</span><span class="identifier" id="mozilla.components.feature.session.SessionUseCases.RequestDesktopSiteUseCase$invoke(kotlin.Boolean, mozilla.components.browser.session.Session)/enable">enable</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/index.html"><span class="identifier">Boolean</span></a><span class="symbol">, </span><span class="identifier" id="mozilla.components.feature.session.SessionUseCases.RequestDesktopSiteUseCase$invoke(kotlin.Boolean, mozilla.components.browser.session.Session)/session">session</span><span class="symbol">:</span>&nbsp;<span class="identifier">Session</span>&nbsp;<span class="symbol">=</span>&nbsp;sessionManager.selectedSessionOrThrow<span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></a></a><br/>
```
Comment (pocmo) [inline docs/api/0.21/feature-session/index-outline.html]: Oh, did the IDE update those files? We don't want to do that for past releases :)

Comment (Amejia481) [reply docs/api/0.21/feature-session/index-outline.html]: Thanks for catching that, I will undo those files

File: components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt [hunk 2]
```diff
@@ -47,6 +47,7 @@ class Session(
         fun onTrackerBlocked(session: Session, blocked: String, all: List<String>) = Unit
         fun onLongPress(session: Session, hitResult: HitResult): Boolean = false
         fun onFindResult(session: Session, result: FindResult) = Unit
+        fun onDesktopModeEnabled(enabled: Boolean) = Unit
```
[the remark this answers is not in the document]
Comment (csadilek) [reply components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt]: Other than that, it looks good to me!

[the remark this answers is not in the document]
Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt]: :thumbsup:

**Comment by Amejia481** [comment]
@csadilek I believe we are ready to go!

**Pull request merged**